### PR TITLE
Refactor / Renaming

### LINKED ISSUE #58: Rename structs and modules to make more sense.
## Modules
- **`bus`** --> **`async_bus`**
- **`channel`** --> **`bus`**
## Structs
- **`Sender`** --> **`Publisher`**
- **`Receiver`** --> **`Subscriber`**
- **`Publisher`** --> **`AsyncPublisher`**
- **`Subscriber`** --> **`AsyncSubscriber`**
- **`Channel`** --> **`RingBuffer`**
## Functions
**`raw_bounded`** --> **`bounded`**
**`bounded`** --> **`async_bounded`**

### PR DESCRIPTION
**Pull request by filipdulic**
Closes #58
## Structs
- Channel -> RingBuffer
- Sender -> Publisher
- Receiver -> Subscriber
- Publisher -> AsyncPublisher
- Subscriber -> AsyncSubscriber
## Modules
- channel::sender -> publisher
- channel::receiver -> subscriber
- bus::publisher -> async_publisher
- bus::subscriber -> async_subscriber
## Moves
- channel::bounded -> bounded
- bus::bounded -> async_bounded
- channel/mod.rs to ring_buffer.rs
- channel/publisher.rs to publisher.rs
- channel/subscriber.rs to subscriber.rs
- bus/publisher.rs to async_publisher.rs
- bus/subscriber.rs to async_subscriber.rs
- tests from bus/mod.rs to tests/async_tests.rs

### CODE REVIEW

File: src/publisher.rs [hunk 1]
```diff
@@ -1,14 +1,14 @@
-use crate::channel::{Channel, SendError};
+use crate::ring_buffer::{RingBuffer, SendError};
 use crate::swap_slot::SwapSlot;
 use std::sync::Arc;
 
 #[derive(Debug)]
-pub struct Sender<T, S: SwapSlot<T>> {
+pub struct GenericPublisher<T, S: SwapSlot<T>> {
```
Comment (stevanmilic) [inline src/publisher.rs]: Why did you call it `GenericPublisher` and not just `Publisher`? Same for `Subscriber`?

File: src/publisher.rs [hunk 2]
```diff
@@ -1,14 +1,14 @@
...
     /// Shared reference to the channel
-    pub(super) channel: Arc<Channel<T, S>>,
+    pub(super) channel: Arc<RingBuffer<T, S>>,
```
Comment (stevanmilic) [inline src/publisher.rs]: I think it makes more sense to call this field `buffer`?

**Review by stevanmilic — CHANGES_REQUESTED** [review]

File: src/publisher.rs (hunk 1 above)
Comment (filipdulic) [reply src/publisher.rs]: fixed in latest commit

File: src/publisher.rs (hunk 2 above)
Comment (filipdulic) [reply src/publisher.rs]: nice catch. fixed in latest commit

**Pull request merged**